### PR TITLE
Fix/users can now go back freely between step 1 and 2 

### LINKED ISSUE #22: Test and verify that users can’t advance without completing each step, but can go backward freely.
@HolyShaq @dssrts @seegyulater check `sidebar.tsx` and resolve the logic for step completion yung issue parang visual issue lang ata idk, thanks!

### PR DESCRIPTION
**Pull request by dssrts**
fixes #22 

before: if uploaded na yung csv, then proceed to step 2, then click step 1 ulit, hindi ka na ulit makakabalik sa step 2 unless iclick mo next
after/fix: user can now go back and forth between step 1 and 2, provided na may uploaded csv file sha, if walang csv, disabled step 2

https://github.com/user-attachments/assets/54e9d11f-e733-482b-806e-a73dbdba02e9

files modified:
`Sidebar.tsx` - added `isCsvUploaded `property to `SidebarProps`
`page.tsx` - added `isCsvUploaded `argument to `Sidebar`
basically added a checker if uploaded na yung csv file

**issue**:
if user advances to step 3, logic is broken and nasisira na sila completely (kahit walang csv nakakapag jump sila ng steps)

### CODE REVIEW

**Review by Gidsss — APPROVED** [review]
lgtm! pls solve the issue in another branch, thanks!
